src/botMessage/dialogFlow.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@nestjs/common';
+const dialogflow = require('dialogflow'); // eslint-disable-line
+
+@Injectable()
+export class DialogFlowService {
+  async getDialogflowIntent(
+    conversationId: string,
+    userMessage: string,
+  ): Promise<string> {
+    let privateKey;
+    if (process.env.NODE_ENV === 'production'){
+      privateKey = JSON.parse(process.env.DIALOGFLOW_PRIVATE_KEY)
+    } else if (process.env.NODE_ENV === 'travis') {
+      privateKey = process.env.DIALOGFLOW_PRIVATE_KEY.replace(/\\n/gm, '\n')
+    } else {
+      process.env.DIALOGFLOW_PRIVATE_KEY
+    }
```

Comment:
This processing will happen every time this method is called – which may be fine, if it's only a few times. But if it's a regular method call, I'd suggest pulling this out to the top so it only happens once, on loading this file.

(Also note, for the `else` branch it doesn't look like `privateKey` is being set?)

Reply:
Updated.